src/wam2layers/tracking/backtrack.py

```diff
@@ -10,6 +10,66 @@
 from wam2layers.utils.profiling import ProgressTracker
 
 
+def reduce_domain(da, config):
+    assert np.min(da.longitude.values) >= 0, "This approach assumes the raw/default longitudes run from 0-360."
+
+    # get the config domain bounds specified by the user
+    bounds = config.reduced_domain_bounds  # [south_lat, north_lat, left_lon, right_lon]
+    south_lat, north_lat = bounds[0], bounds[1]
+    left_lon, right_lon = bounds[2], bounds[3]
+
+    # check if need to map longitude to -180 to 180 for better definition of the domain
+    if left_lon > right_lon:
+        da.coords["longitude"] = (da.coords["longitude"] + 180) % 360 - 180
+        da = da.sortby(da.longitude)
+        da = da.transpose(..., "latitude", "longitude")
+
+        left_lon = (left_lon + 180.) % 360. - 180.
+        right_lon = (right_lon + 180.) % 360. - 180.
+        assert left_lon < right_lon, "Something is not right. Try flipping the order of your longitude bounds."
+
+        check_periodic_boundaries(longitude_units="180_180", config=config)
+    else:
+        check_periodic_boundaries(longitude_units="0_360", config=config)
+
+    mask_lon = (da.longitude >= left_lon) & (da.longitude <= right_lon)
+    mask_lat = (da.latitude >= south_lat) & (da.latitude <= north_lat)
+    return da.where(mask_lon & mask_lat, drop=True)
```

Comment:
I'm wondering if you could alternatively do 
```suggestion
    return da.sel(longitude=Slice(left_lon, right_lon), latitude=Slice(south_lat, north_lat))
```
I think that should also work for a regular lat/lon grid (haven't tested though), and I think it could be more memory efficient.

Reply:
I think this switch is good. Slicing in this way is inclusive I believe, so should repeat my code exactly. (this replacement won't work if it is an OR, instead of an AND...but that is not relevant for this code snippet).